internal/benchmarks/benchmarks_test.go

```diff
@@ -0,0 +1,199 @@
+				require.NoError(b, mig.Complete(ctx))
+				b.StopTimer()
+				b.Logf("Backfilled %d rows in %s", rowCount, b.Elapsed())
+				rowsPerSecond := float64(rowCount) / b.Elapsed().Seconds()
+				b.ReportMetric(rowsPerSecond, unitRowsPerSecond)
+			})
+		})
+	}
+}
+
+// Benchmark the difference between updating all rows with and without an update trigger in place
+func BenchmarkWriteAmplification(b *testing.B) {
+	ctx := context.Background()
+	testSchema := testutils.TestSchema()
+	var opts []roll.Option
+
+	assertRowCount := func(tb testing.TB, db *sql.DB, rowCount int) {
+		tb.Helper()
+
+		var count int
+		err := db.QueryRowContext(ctx, "SELECT COUNT(*) FROM users WHERE name = 'person'").Scan(&count)
+		require.NoError(b, err)
+		require.Equal(b, rowCount, count)
+	}
+
+	b.Run("NoTrigger", func(b *testing.B) {
+		for _, rowCount := range rowCounts {
+			b.Run(strconv.Itoa(rowCount), func(b *testing.B) {
+				testutils.WithMigratorInSchemaAndConnectionToContainerWithOptions(b, testSchema, opts, func(mig *roll.Roll, db *sql.DB) {
+					setupInitialTable(b, ctx, testSchema, mig, db, rowCount)
+					b.Cleanup(func() {
+						require.NoError(b, mig.Close())
+						assertRowCount(b, db, rowCount)
+					})
+
+					b.ResetTimer()
+
+					// Update the name in all rows
+					b.StartTimer()
+					_, err := db.ExecContext(ctx, `UPDATE users SET name = 'person'`)
+					require.NoError(b, err)
+					b.StopTimer()
+					rowsPerSecond := float64(rowCount) / b.Elapsed().Seconds()
+					b.ReportMetric(rowsPerSecond, unitRowsPerSecond)
+				})
+			})
+		}
+	})
+
+	b.Run("WithTrigger", func(b *testing.B) {
+		for _, rowCount := range rowCounts {
+			b.Run(strconv.Itoa(rowCount), func(b *testing.B) {
+				testutils.WithMigratorInSchemaAndConnectionToContainerWithOptions(b, testSchema, opts, func(mig *roll.Roll, db *sql.DB) {
+					setupInitialTable(b, ctx, testSchema, mig, db, rowCount)
+
+					// Start the migration
+					require.NoError(b, mig.Start(ctx, &migAlterColumn))
+					b.Cleanup(func() {
+						// Finish the migration
+						require.NoError(b, mig.Complete(ctx))
+						require.NoError(b, mig.Close())
+						assertRowCount(b, db, rowCount)
+					})
+
+					b.ResetTimer()
+
+					// Update the name in all rows
+					b.StartTimer()
+					_, err := db.ExecContext(ctx, `UPDATE users SET name = 'person'`)
+					require.NoError(b, err)
+					b.StopTimer()
+					rowsPerSecond := float64(rowCount) / b.Elapsed().Seconds()
+					b.ReportMetric(rowsPerSecond, unitRowsPerSecond)
+				})
+			})
+		}
+	})
+}
+
+func setupInitialTable(tb testing.TB, ctx context.Context, testSchema string, mig *roll.Roll, db *sql.DB, rowCount int) {
+	tb.Helper()
+
+	seed := func(tb testing.TB, rowCount int, db *sql.DB) {
+		seedStart := time.Now()
+		defer func() {
+			elapsed := time.Since(seedStart)
+			rowsPerSecond := float64(rowCount) / elapsed.Seconds()
+			tb.Logf("Seeded %d rows in %s (%.f rows/s)", rowCount, elapsed, rowsPerSecond)
```

Comment:
Can be removed - I don't think the seed duration is relevant to report as it's outside the control of `pgroll`.